### PR TITLE
add optional persistence to chartmuseum and upgrade image to latest r…

### PR DESCRIPTION
**Pull request by rawlingsj**
…elease

- upgrade chartmuseum image to v0.2.4
- add a PVC to persist the storage folder so charts published charts surive pod restarts
- add option to disable persistance

### CODE REVIEW

**Comment by rawlingsj** [comment]
I've just incremented the chart version for CI to pass - is there anything else missing before we can trigger the CI tests for this change?

File: incubator/chartmuseum/values.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 replicaCount: 1
 image:
   repository: chartmuseum/chartmuseum
-  tag: v0.1.0
+  tag: v0.2.4
```
Comment (jdolitsky) [inline incubator/chartmuseum/values.yaml]: Can we bump this to v0.2.6?

File: incubator/chartmuseum/Chart.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 description: Helm Chart Repository with support for Amazon S3 and Google Cloud Storage
 name: chartmuseum
-version: 0.1.0
+version: 0.1.1
 appVersion: 0.1.0
```
Comment (jdolitsky) [inline incubator/chartmuseum/Chart.yaml]: Can we change `appVersion` to 0.2.6?

**Review by jdolitsky — CHANGES_REQUESTED** [review]
Nice! Did not see this come in. Can we update version to 0.2.6?

**Comment by jdolitsky** [comment]
Also looks like the lint failed:
```
==> Linting incubator/chartmuseum
[ERROR] templates/: render error in "chartmuseum/templates/service.yaml": template: chartmuseum/templates/service.yaml:4:11: executing "chartmuseum/templates/service.yaml" at <include "common.full...>: error calling include: template: no template "common.fullname" associated with template "gotpl"

Error: 1 chart(s) linted, 1 chart(s) failed

Linting the Chart.yaml and values.yaml files at incubator/chartmuseum
incubator/chartmuseum/values.yaml
  33:22     error    no new line character at the end of file  (new-line-at-end-of-file)
```

Seems as though the script isnt grabbing the common dependency before running lint..

File: incubator/chartmuseum/values.yaml [hunk 3]
```diff
@@ -14,3 +14,20 @@ resources:
   requests:
     cpu: 80m
     memory: 64Mi
+persistence:
+  Enabled: true
```
Comment (davidovich) [inline incubator/chartmuseum/values.yaml]: As others would want to customize to their needs, I think this should be defaulted to `false`. Deciding to use `true` here would favor the use of pvcs over other storage mediums like object storage. There will only ever be one storage active at a time. What do you think ?

**Comment by jdolitsky** [comment]
@rawlingsj can we please do the following:
- rebase from master to pick up changes to CI scripts (#2934)
- add newline to values.yaml to fix lint.sh
- update to latest chartmuseum version (v0.2.6)
- disable persistence by default, as suggested by @davidovich

Any other suggestions?

@MichielDeMey @arodus @joosthof @SlickNik @stefsuse - lets try to get this one merged. If #2852 or #2900 provides additional value, please suggest here or wait to rebase on top of this change. Otherwise please close duplicate PRs.

File: incubator/chartmuseum/Chart.yaml (hunk 2 above)
Comment (rawlingsj) [reply incubator/chartmuseum/Chart.yaml]: sure thing, done

File: incubator/chartmuseum/values.yaml (hunk 1 above)
Comment (rawlingsj) [reply incubator/chartmuseum/values.yaml]: sure thing, done

File: incubator/chartmuseum/values.yaml (hunk 3 above)
Comment (rawlingsj) [reply incubator/chartmuseum/values.yaml]: yeah makes sense, done

**Comment by rawlingsj** [comment]
thanks @jdolitsky @davidovich - comments address although I didn't totally grok the lint failure as it seemed to work here, I added a newline at the end of values.yaml hoping that should sort it anyways.

**Comment by rawlingsj** [comment]
Ah I see the circleci failure now, ok lemme take a look.

**Review by jdolitsky — APPROVED** [review]

**Pull request merged**